src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -124,7 +121,12 @@ class LazyVals extends MiniPhaseTransform with SymTransformer {
 
 
   override def transformStats(trees: List[tpd.Tree])(implicit ctx: Context, info: TransformerInfo): List[tpd.Tree] = {
-    val (holders, stats) = trees.partition { _.symbol.flags == containerFlags}
+    val (holders, stats) =
+      atGroupEnd { implicit ctx: Context =>
+        trees.partition {
+          _.symbol.flags == containerFlags
```

Comment:
They are reordered. Otherwise backend fails to compile methods with forward references to variables.
I'll add a comment.